btc_execution/sources/interpreter.move

```diff
@@ -69,18 +83,74 @@ fun cast_to_bool(v: &vector<u8>): bool {
 }
 
 
+fun op_equal(ip: &mut Interpreter) {
+    let first_value = ip.stack.pop();
+    let second_value = ip.stack.pop();
+    let ans = if (first_value == second_value) {
+        vector[1]
+    } else {
+        vector[0]
+    };
+    ip.stack.push(ans);
+}
+
+fun op_equal_verify(ip:&mut Interpreter) {
+    ip.op_equal();
+    assert!(ip.stack.pop() == vector[1], EEqualVerify);
```

Comment:
```suggestion
    assert_eq!(ip.stack.pop(), vector[1], EEqualVerify);
```

Reply:
assert_eq! doesn't support returning an error code. We need to use assert!